util/util.go

```diff
@@ -288,6 +288,23 @@ func SafeClose(closer func() error, err *error) {
 	}
 }
 
+// SafeCloseDeferLogger takes a file and runs the deferred closing of both. Used for custom agent uploads
+func SafeCloseDeferLogger(file *os.File) {
+	defer func(name string) {
```

Comment:
Deferring things within a method doesn't do much for you.

In this case consider having the two operations run normally (close -> defer), and defer this function from the caller.